### PR TITLE
Refactor Update Checker

### PR DESCRIPTION
**Pull request by duk3luk3**
`GithubUpdateChecker` -> `UpdateChecker`

Collects Github stable release, github prerelease, and server update
info

For #671 

- [x] PR branch should be named `issuenum`-`fix`/`feature`/`cleanup`-`description`
- [x] Code is split into logical commits
- [ ] Code has tests
- [x] Final commit includes "Fixes #issue" in commit message

When all builds pass and a maintainer is happy with your PR, the "ready" label will be applied. Please complete these tasks then:

- [x] Rebase onto develop
- [x] Add changelog entry
- [ ] Remove this entire template section

### CODE REVIEW

File: src/client/updater.py [hunk 1]
```diff
@@ -15,33 +15,64 @@
...
+        self._req_done()
+
+    def server_session(self):
+        self._server_info = {}
+        self._req_done()
+
+
+    def _parse_release(self, release_info):
+            try:
+                js = json.loads(release_info))[0]
```
Comment (Wesmania) [inline src/client/updater.py]: s@))@)@

**Comment by muellni** [comment]
Looks good. 
I'm having a small problem on my machine: The `update_settings.ui` dialog can be resized such that the `lblExplanation` text is truncated. IDK if that happens on Windows too.

**Comment by muellni** [comment]
https://github.com/duk3luk3/client/pull/2 should fix this (and renames the dialog to "Update Settings")

**Comment by muellni** [comment]
and here's another little fix: https://github.com/duk3luk3/client/pull/3

**Pull request merged**